test/regression/sql/prepare.sql

```diff
@@ -0,0 +1,54 @@
+-- the view should return the empty
+SELECT name, statement, parameter_types FROM pg_prepared_statements;
+
+CREATE DATABASE testdb;
+CREATE TABLE copy_database(datname text, datistemplate boolean, datallowconn boolean);
+INSERT INTO copy_database SELECT datname, datistemplate, datallowconn FROM pg_database;
+-- parameterized queries
+PREPARE q1(text) AS
+	SELECT datname, datistemplate, datallowconn
+	FROM copy_database WHERE datname = $1;
+
+EXECUTE q1('testdb');
+
+-- q1
+SELECT name, statement, parameter_types FROM pg_prepared_statements
+    ORDER BY name;
+
+CREATE TABLE ta(a int);
+INSERT INTO ta(a) SELECT * FROM generate_series(1, 1000);
+SELECT COUNT(*) FROM ta;
+PREPARE q2 AS SELECT COUNT(*) FROM ta;
+EXECUTE q2;
+
+-- q1 q2
+SELECT name, statement, parameter_types FROM pg_prepared_statements
+    ORDER BY name;
+
+PREPARE q3(int) AS
+	SELECT * FROM ta WHERE a = $1;
+
+CREATE TEMPORARY TABLE q3_prep_results AS EXECUTE q3(200);
+SELECT * FROM q3_prep_results;
+
+-- q1 q2 q3
+SELECT name, statement, parameter_types FROM pg_prepared_statements
+    ORDER BY name;
+
+CREATE TABLE tb (a int DEFAULT 1, b int, c varchar DEFAULT 'pg_duckdb');
+INSERT INTO tb(b) VALUES(2);
+PREPARE q4(int, varchar) AS
+	SELECT b FROM tb WHERE a = $1 AND c = $2;
+EXECUTE q4(1, 'pg_duckdb');
```

Comment:
I'd like to see a query too where the same parameter is used multiple times

Reply:
done

Reply:
I think you misunderstood what I meant. I didn't mean that the statement should be executed multiple times. I meant that we should have a query where the same parameter is used multiple times.

```
PREPARE q5(int, varchar) AS
	SELECT b FROM tb WHERE a = $1 AND b = $1 AND c = $2;
```

And I think it would be good to also check versions where one of the parameters is not used (both the first parameter and the last parameter unused is interesting I think):
```
PREPARE q6(int, varchar) AS
	SELECT b FROM tb WHERE c = $2;
PREPARE q7(int, varchar) AS
	SELECT b FROM tb WHERE a = $1;
```

Reply:
Related test cases have been added, I really misunderstood your meaning, thank you for your guidance